packages/engine-core/src/data-proxy/utils/request.ts

```diff
@@ -38,7 +38,8 @@ export async function request(url: string, options: RequestOptions = {}): Promis
  */
 function buildHeaders(options: RequestOptions): RequestOptions['headers'] {
   return {
-    ...options.headers,
+    // this ensures headers will always be valid
+    ...JSON.parse(JSON.stringify(options.headers)),
```

Comment:
Which value was passed that prompted this? 👀

Reply:
`undefined`

Reply:
Discarded correctly everywhere besides the custom nodeFetch

Reply:
I see https://github.com/prisma/e2e-tests/pull/2379#issuecomment-1026656733